src/Microsoft.Azure.WebJobs.Host/InvokeFunctionFilterAttribute.cs

```diff
@@ -0,0 +1,128 @@
+        {
+            if (executingContext == null)
+            {
+                throw new ArgumentNullException("executingContext");
+            }
+            
+            if (!string.IsNullOrEmpty(ExecutingFilter))
+            {
+                executingContext.Logger.LogInformation("Executing Function Filter '" + ExecutingFilter + "'");
+
+                try
+                {
+                    await InvokeJobFunctionAsync(ExecutingFilter, executingContext, cancellationToken);
+                }
+                catch (Exception e)
+                {
+                    executingContext.Logger.LogInformation(e.ToDetails());
+                }
+            }
+            else
+            {
+                await base.OnExecutingAsync(executingContext, cancellationToken);
+            }
+        }
+
+        /// <summary>
+        /// Call the requested job method after the main function call
+        /// </summary>
+        /// <returns></returns>
+        public override async Task OnExecutedAsync(FunctionExecutedContext executedContext, CancellationToken cancellationToken)
+        {
+            if (executedContext == null)
+            {
+                throw new ArgumentNullException("executingContext");
+            }
+
+            if (!string.IsNullOrEmpty(ExecutedFilter))
+            {
+                executedContext.Logger.LogInformation("Executing Function Filter '" + ExecutedFilter + "'");
+
+                try
+                {
+                    await InvokeJobFunctionAsync(ExecutedFilter, executedContext, cancellationToken);
+                }
+                catch (Exception e)
+                {
+                    executedContext.Logger.LogInformation(e.ToDetails());
+                }
+            }
+            else
+            {
+                await base.OnExecutedAsync(executedContext, cancellationToken);
+            }
+        }
+
+        internal async Task InvokeJobFunctionAsync<TContext>(string methodName, TContext context, CancellationToken cancellationToken) where TContext : FunctionInvocationContext
+        {
+            MethodInfo methodInfo = null;
+
+            foreach (var type in context.Config.TypeLocator.GetTypes())
+            {
+                if (methodName != null)
+                {
+                    methodInfo = type.GetMethod(methodName, BindingFlags.Public | BindingFlags.Instance | BindingFlags.Static);
+                }
+
+                if (methodInfo != null)
+                {
+                    break;
+                }
+            }
+
+            // If it is null, this means that we need to look for the methodinfo from the class
+            // TODO: Implement class level methodinfo onExecuting/onExecuted invocation
+
+            IDictionary<string, object> invokeArguments = new Dictionary<string, object>();
+            string parameterName = methodInfo.GetParameters().SingleOrDefault(p => p.ParameterType == typeof(TContext)).Name;
```

Comment:
Also, the parameter type check needs to be changed to p.ParameterType.IsAssignableFrom(typeof(TContext))